src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java

```diff
@@ -46,14 +59,103 @@ private boolean checkIfProductsCacheNeedsUpdate() {
+        BigDecimal priceLow = new BigDecimal("0");
+        BigDecimal priceHigh = sortedProducts.get(sortedProducts.size() - 1).getPrice();
+        searchService.setPriceLow(priceLow);
+        searchService.setPriceHigh(priceHigh);
+    }
+
+    @Override
+    public ProductList filterByPrice(List<Product> products, SearchModel searchModel) {
+        setAppliedFiltersToSearchModel(searchModel);
+        List<Product> result = products
+                .parallelStream()
+                .filter(applyPriceFilter(searchModel))
+                .collect(Collectors.toUnmodifiableList());
+        return getSearchResultsOrAllProducts(result);
+    }
+
+    private ProductList getSearchResultsOrAllProducts(List<Product> result) {
+        searchService.setSearchFailure(result.isEmpty());
+        return result.isEmpty()
+                ? new ProductList(getProductListCache().getProducts())
+                : new ProductList(result);
+    }
+
+    private void setAppliedFiltersToSearchModel(SearchModel searchModel) {
+        searchModel.setAppliedFiltersHeader(" with the applied filters");
+        String searchRequest = (null != searchService.getSearchModel().getSearchRequest())
+                ? searchService.getSearchModel().getSearchRequest()
+                : "";
+        searchModel.setSearchRequest(searchRequest);
+        searchModel.setSearchFailure(false);
+        searchModel.setActiveFilters(true);
+    }
+
+    private Predicate<Product> applyPriceFilter(SearchModel searchModel) {
+        return product -> product.getPrice().compareTo(searchModel.getPriceLow()) >= 0 && product.getPrice().compareTo(searchModel.getPriceHigh()) <= 0;
+    }
+
+    @Override
+    public ProductList sortListByName(List<Product> products, SearchModel searchModel) {
+        final Comparator<Product> productNameComparator = (searchModel.isSortAscendingName())
+                ? Comparator.comparing(Product::getName)
+                : Comparator.comparing(Product::getName).reversed();
+        return getSortedList(products, productNameComparator);
+    }
+
+    @Override
+    public ProductList sortListByPrice(List<Product> products, SearchModel searchModel) {
+        final Comparator<Product> productPriceComparator = (searchModel.isSortAscendingPrice())
+                ? Comparator.comparing(Product::getPrice)
+                : Comparator.comparing(Product::getPrice).reversed();
+        return getSortedList(products, productPriceComparator);
+    }
+
+    private ProductList getSortedList(List<Product> products, Comparator<Product> productComparator) {
+        return new ProductList(
+                products
+                        .stream()
+                        .sorted(productComparator)
+                        .collect(Collectors.toUnmodifiableList()));
+    }
+
+    private List<Product> executeSearch(String searchRequest) {
+        return (null != searchRequest)
```

Comment:
boolean hasSearchRequest(searchRequest) is more readable in this if statement.

Reply:
I disagree. Whilst I get the logic behind extracting every little boolean-check, I myself wouldn't do so for simple null-checks, especially this one. Here's why: 
Above, I've extracted a method checking for searchRequest. Hence, the method already exists. Doing the exact same thing here, would indeed result in the method you suggest. BUT, that would mean that my previous extracted boolean-check would become something like -> hasSearchRequest(searchService.getSearchModel().getSearchRequest()).
If we're looking for increased readability, this isn't the way to go...